### ISSUE TITLE
[Feature-Request]: Show variable info when hovering variables of variable lists

### ISSUE THREAD

**Issue by keunes** [jasp-stats/jasp-issues#1190]
* Enhancement: Import the question titles/labels when opening sav file, instead of the question code
* Purpose: Make analysis easier
* Use-case: Importing an sav file from SurveyMonkey and run basic stats

**Describe the solution you'd like**
 
Automatically pick up the questions, just as happens when importing a CSV file.

**Additional context**

As noted the sav file comes from SurveyMonkey - I'm not sure it actually actually contains the questions (though that would surprise me). I tried opening in a text editor, but that provides garble. I tried to edit the data (supposedly the questions can be edited there) but ran into #1189.

**Comment by boutinb** [comment]
Can't you save the file in csv format from SPSS? I thought the labels were also loaded in JASP, could you send me your save file?

**Comment by keunes** [comment]
@boutinb I have three formats that I downloaded from SurveyMonkey, and indeed I could open a sav file and save it as a csv.
The thing is:
* the sav has numerical values and the labels added, but doesn't currently contain the questions
![image](https://user-images.githubusercontent.com/11229646/111799799-a309d000-88cb-11eb-92a9-09bd46803568.png)
* the two csv versions contain the questions, but the values and labels are the same (both numbers or both strings).

I would like the combo of having numerical values, with the labels (for ordinal/nominal variables), and also the questions. Now, I am not sure, but I would expect the questions are available in the SPSS file that SurveyMonkey exports. But maybe I'm wrong and SurveyMonkey does not include the question texts in the sav export? (which would be odd, given that it _does_ include the texts in the csv export).

> could you send me your save file?

I'm afraid not - we included a prize draw so the dataset contains personal data. And I don't think it's posible to manipulate which columns (questions) are included in the download.
Do you know if a sav file can be 'inspected' in any other way, to see if it might somewhere contain the question texts?

If not, next week I could do a dummy survey in SurveyMonkey, reply a few times, and export the sav file to share with you.

**Comment by keunes** [comment]
I just managed to get an SPSS trial installed, and opened the same data file.

The file does contain the question labels:
![image](https://user-images.githubusercontent.com/11229646/111994787-739bd300-8b18-11eb-9c48-7d7cac5a0ea3.png)

But in the dataset view, SPSS also just display the question codes:
![image](https://user-images.githubusercontent.com/11229646/111994940-aa71e900-8b18-11eb-8948-a7d987f9fbcb.png)

Maybe my request should be changed to "Display question label in the Analysis screen"

That's what SPSS does, and it makes analysis a whole lot easier:
![image](https://user-images.githubusercontent.com/11229646/111996289-26206580-8b1a-11eb-9b03-7b0ae203128c.png)

Currently in JASP:
![image](https://user-images.githubusercontent.com/11229646/111995806-a4c8d300-8b19-11eb-8e7b-32d6de61f69a.png)

**Comment by boutinb** [comment]
Hi @keunes, a big disadvantage with question label, is that they can be quite long, making it quite unclear in a Variables List to see exactly which question it is: as the variable name must be cut to fit the width of the list, and as many questions may start with the same words, many variable names would look the same (only if your cursor is on it, the whole question is then displayed in a tooltip).
One solution would be to import both question names & labels, so that the names are displayed in the variable list, but the labels are displayed in the tooltip (when the cursor is on a variable). But that can be quite some work to implement. 
Another easier solution would be to have a setting somewhere to specify to import the question labels in place of the question names in a sav file: you would have the disadvantage to have this long names, but you would have at least the choice.

**Comment by keunes** [comment]
> One solution would be to import both question names & labels, so that the names are displayed in the variable list, but the labels are displayed in the tooltip (when the cursor is on a variable).

I think that should be the way to go. Personally I typically prefer waiting longer for a proper fix, than waiting shorter for a workaround :)

**Comment by tomtomme** [comment]
@JorisGoosen & @keunes 
Since data edit mode we have the "Description" field for every variable  - do we already import the descriptions from SAV files there? Then it should be easy to show a popup, that shows the full description, when one is hovering above a variable e.g. within the variable list in computed columns or within the variable list within an analysis. Or is this harder than I imagine?

We also have the "Long name" field. I do not think that SPSS has an equivalent for this and wonder where and how it could be used. Maybe it would be nice to be able to switch between "Name" and "Long name" in variable lists or even in data edit mode etc.

**Comment by JorisGoosen** [comment]
We do not yet get this information from spss files.

Adding the short and long label would probably not be too much work.
But before we get on that we are going to work on some more refactoring of the underlying code for the variables.

Like for instance https://github.com/jasp-stats/jasp-desktop/pull/5367
Once that is done I can have a look at https://github.com/jasp-stats/jasp-issues/issues/2500 and getting as much info from readstat as possible.

**Comment by tomtomme** [comment]
@keunes 
We now have such a pop-up in data-view via 0.19 beta:
![Screenshot_20240527_131516](https://github.com/jasp-stats/jasp-issues/assets/4428185/09b75516-c6bb-43cb-b967-d8d5d49ce5fb)

However there is no such popup yet in the analysis-view, so I will mark this one for 0.20. Maybe this it can be included then.

**Comment by JorisGoosen** [comment]
Would be good to inventarise where one would need to see these tooltips.

- Variables list/forms
- filter/comp.column constructors
- ?

**Comment by tomtomme** [comment]
Thats it already. Showing them in the results / tables / plots is not as important because there the process of finding the right vars to analyse is already done
